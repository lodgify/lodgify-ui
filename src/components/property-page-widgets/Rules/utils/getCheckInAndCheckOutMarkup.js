import React, { Fragment } from 'react';

import { Icon, ICON_NAMES } from 'elements/Icon';
import { GridColumn } from 'layout/GridColumn';
import { Divider } from 'elements/Divider';

import { getLabelAndValueString } from './getLabelAndValueString';

/**
 * @param {string} checkInTimeLabel
 * @param {string} checkInTime
 * @param {string} checkOutTimeLabel
 * @param {string} checkOutTime
 * @return {Object}
 */
export const getCheckInAndCheckOutMarkup = (
  checkInTimeLabel,
  checkInTime,
  checkOutTimeLabel,
  checkOutTime
) =>
  (!!checkInTime || !!checkOutTime) && (
    <GridColumn computer={9} tablet={7}>
      {!!checkInTime && (
        <Fragment>
          <Icon
            labelText={getLabelAndValueString(checkInTimeLabel, checkInTime)}
            name={ICON_NAMES.CHECK_IN}
          />
          <Divider />
        </Fragment>
      )}
      {!!checkOutTime && (
        <Icon
          labelText={getLabelAndValueString(checkOutTimeLabel, checkOutTime)}
          name={ICON_NAMES.CHECK_OUT}
        />
      )}
    </GridColumn>
  );
