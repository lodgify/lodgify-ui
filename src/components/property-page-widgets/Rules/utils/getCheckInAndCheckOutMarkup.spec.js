import { getCheckInAndCheckOutMarkup } from './getCheckInAndCheckOutMarkup';

const checkInLabel = 'The Label';
const checkInValue = '9:00 AM';
const checkOutLabel = 'The Label';
const checkOutValue = '9:00 AM';

describe('`getCheckInAndCheckOutMarkup`', () => {
  describe('if all check in and check out values are truthy', () => {
    it('should render the right structure', () => {
      const actual = getCheckInAndCheckOutMarkup(
        checkInLabel,
        checkInValue,
        checkOutLabel,
        checkOutValue
      );

      expect(actual).toMatchSnapshot();
    });
  });

  describe('if any check in value is false or undefined', () => {
    it('should render the right structure', () => {
      const actual = getCheckInAndCheckOutMarkup(
        checkInLabel,
        '',
        checkOutLabel,
        checkOutValue
      );

      expect(actual).toMatchSnapshot();
    });
  });

  describe('if any check out value is false or undefined', () => {
    it('should render the right structure', () => {
      const actual = getCheckInAndCheckOutMarkup(
        checkInLabel,
        checkInValue,
        undefined,
        checkOutValue
      );

      expect(actual).toMatchSnapshot();
    });
  });
  describe('if any check in or check out value is false or undefined', () => {
    it('should return false', () => {
      const actual = getCheckInAndCheckOutMarkup(
        checkInLabel,
        '',
        checkOutLabel,
        undefined
      );

      expect(actual).toMatchSnapshot();
    });
  });
});
