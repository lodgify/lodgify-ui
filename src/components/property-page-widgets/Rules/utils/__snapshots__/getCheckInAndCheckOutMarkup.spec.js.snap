// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`\`getCheckInAndCheckOutMarkup\` if all check in and check out values are truthy should render the right structure 1`] = `
<GridColumn
  computer={9}
  tablet={7}
  verticalAlignContent="top"
  width={null}
>
  <React.Fragment>
    <Icon
      color={null}
      hasBorder={false}
      isButton={false}
      isCircular={false}
      isColorInverted={false}
      isDisabled={false}
      isLabelLeft={false}
      labelText="The Label: 9:00 AM"
      labelWeight={null}
      name="check in"
      path={null}
      size={null}
    />
    <Divider
      className=""
      hasLine={false}
      size="medium"
    />
  </React.Fragment>
  <Icon
    color={null}
    hasBorder={false}
    isButton={false}
    isCircular={false}
    isColorInverted={false}
    isDisabled={false}
    isLabelLeft={false}
    labelText="The Label: 9:00 AM"
    labelWeight={null}
    name="check out"
    path={null}
    size={null}
  />
</GridColumn>
`;

exports[`\`getCheckInAndCheckOutMarkup\` if any check in or check out value is false or undefined should return false 1`] = `false`;

exports[`\`getCheckInAndCheckOutMarkup\` if any check in value is false or undefined should render the right structure 1`] = `
<GridColumn
  computer={9}
  tablet={7}
  verticalAlignContent="top"
  width={null}
>
  <Icon
    color={null}
    hasBorder={false}
    isButton={false}
    isCircular={false}
    isColorInverted={false}
    isDisabled={false}
    isLabelLeft={false}
    labelText="The Label: 9:00 AM"
    labelWeight={null}
    name="check out"
    path={null}
    size={null}
  />
</GridColumn>
`;

exports[`\`getCheckInAndCheckOutMarkup\` if any check out value is false or undefined should render the right structure 1`] = `
<GridColumn
  computer={9}
  tablet={7}
  verticalAlignContent="top"
  width={null}
>
  <React.Fragment>
    <Icon
      color={null}
      hasBorder={false}
      isButton={false}
      isCircular={false}
      isColorInverted={false}
      isDisabled={false}
      isLabelLeft={false}
      labelText="The Label: 9:00 AM"
      labelWeight={null}
      name="check in"
      path={null}
      size={null}
    />
    <Divider
      className=""
      hasLine={false}
      size="medium"
    />
  </React.Fragment>
  <Icon
    color={null}
    hasBorder={false}
    isButton={false}
    isCircular={false}
    isColorInverted={false}
    isDisabled={false}
    isLabelLeft={false}
    labelText="undefined: 9:00 AM"
    labelWeight={null}
    name="check out"
    path={null}
    size={null}
  />
</GridColumn>
`;
